Done.
Setting up the random seed for reproducibility...
Done.
dynamic_import - class_name:  OracleTrainer
Initializing model...
Done.
Initializing datasets...
Done.
Initializing clients...
dynamic_import - class_name:  OracleClient
Done.
Initializing server...
dynamic_import - class_name:  OracleServer
Running without server optimizer
Done.
Initialize return score, metrics, ckpt, ckpt step...
Setting up metrics...
Done.
Done.
Initializing optimizer and scheduler...
Done.
dynamic_import - class_name:  OracleClient
The experiment begins...
ROUND 1/3: Training 5 Clients...
CLIENT 1/5: erfurt5 RGB
Exception ignored in: <function _MultiProcessingDataLoaderIter.__del__ at 0x7f37e33f2b80>
Traceback (most recent call last):
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 1328, in __del__
    self._shutdown_workers()
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 1296, in _shutdown_workers
    self._mark_worker_as_unavailable(worker_id, shutdown=True)
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 1243, in _mark_worker_as_unavailable
    q.put(None)
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/multiprocessing/queues.py", line 94, in put
    self._start_thread()
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/multiprocessing/queues.py", line 179, in _start_thread
    self._thread.start()
  File "/home/utente/anaconda3/envs/LADD/lib/python3.9/threading.py", line 883, in start
    if not self._initialized:
KeyboardInterrupt:
Traceback (most recent call last):
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 53, in <module>
    run_experiment()
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 21, in run_experiment
    max_score = trainer.train(*trainer.train_args, **trainer.train_kwargs)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/oracle_trainer.py", line 136, in train
    return self.perform_fed_oracle_training(
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/oracle_trainer.py", line 76, in perform_fed_oracle_training
    losses = self.server.train_clients(partial_metric=partial_train_metric)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/servers/oracle_server.py", line 101, in train_clients
    out = c.train(partial_metric, r=r)
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/oracle_client.py", line 199, in train
    dict_all_iters_losses = self.run_epoch(epoch, optimizer, metric=partial_metric, r=r)
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/oracle_client.py", line 169, in run_epoch
    self.__exec_epoch(optimizer, cur_epoch, metric, scheduler, plot_lr, dict_all_iters_losses,
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/oracle_client.py", line 134, in __exec_epoch
    dict_calc_losses, outputs = self.calc_loss_and_output(images, labels)
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/client.py", line 108, in calc_loss_and_output
    else:
KeyboardInterrupt